crates/nostr-mls-memory-storage/src/lib.rs

```diff
@@ -0,0 +1,583 @@
+        let size = NonZeroUsize::new(cache_size)
+            .unwrap_or_else(|| NonZeroUsize::new(DEFAULT_CACHE_SIZE).unwrap());
+
+        NostrMlsMemoryStorage {
+            openmls_storage: storage_implementation,
+            groups_cache: RwLock::new(LruCache::new(size)),
+            groups_by_nostr_id_cache: RwLock::new(LruCache::new(size)),
+            group_relays_cache: RwLock::new(LruCache::new(size)),
+            welcomes_cache: RwLock::new(LruCache::new(size)),
+            processed_welcomes_cache: RwLock::new(LruCache::new(size)),
+            messages_cache: RwLock::new(LruCache::new(size)),
+            messages_by_group_cache: RwLock::new(LruCache::new(size)),
+            processed_messages_cache: RwLock::new(LruCache::new(size)),
+        }
+    }
+}
+
+impl Default for NostrMlsMemoryStorage {
+    /// Creates a new `NostrMlsMemoryStorage` with a default OpenMLS memory storage implementation.
+    ///
+    /// # Returns
+    ///
+    /// A new instance of `NostrMlsMemoryStorage` with default configuration.
+    fn default() -> Self {
+        Self::new(MemoryStorage::default())
+    }
+}
+
+/// Implementation of `NostrMlsStorageProvider` for memory-based storage.
+impl NostrMlsStorageProvider for NostrMlsMemoryStorage {
+    type OpenMlsStorageProvider = MemoryStorage;
+
+    /// Returns the backend type.
+    ///
+    /// # Returns
+    ///
+    /// [`Backend::Memory`] indicating this is a memory-based storage implementation.
+    fn backend(&self) -> Backend {
+        Backend::Memory
+    }
+
+    /// Get a reference to the openmls storage provider.
+    ///
+    /// This method provides access to the underlying OpenMLS storage provider.
+    /// This is primarily useful for internal operations and testing.
+    ///
+    /// # Returns
+    ///
+    /// A reference to the openmls storage implementation.
+    fn openmls_storage(&self) -> &Self::OpenMlsStorageProvider {
+        &self.openmls_storage
+    }
+
+    /// Get a mutable reference to the openmls storage provider.
+    ///
+    /// This method provides mutable access to the underlying OpenMLS storage provider.
+    /// This is primarily useful for internal operations and testing.
+    ///
+    /// # Returns
+    ///
+    /// A mutable reference to the openmls storage implementation.
+    fn openmls_storage_mut(&mut self) -> &mut Self::OpenMlsStorageProvider {
+        &mut self.openmls_storage
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use nostr::{EventId, Kind, PublicKey, RelayUrl, Tags, Timestamp, UnsignedEvent};
+    use nostr_mls_storage::groups::types::{Group, GroupState, GroupType};
+    use nostr_mls_storage::groups::GroupStorage;
+    use nostr_mls_storage::messages::types::{Message, ProcessedMessageState};
+    use nostr_mls_storage::messages::MessageStorage;
+    use nostr_mls_storage::welcomes::types::{ProcessedWelcomeState, Welcome, WelcomeState};
+    use nostr_mls_storage::welcomes::WelcomeStorage;
+    #[cfg(test)]
```

Comment:
Remove this `#[cfg(test)]`, since it's already in a block that uses that condition